### PR TITLE
fixing issue where AzureActiveDirectoryCredentials.AccessToken is null

### PR DESCRIPTION
**Pull request by brettsam**
Fixes Issue #59.

### CODE REVIEW

**Comment by phvannor** [comment]
Would this let me fix the AAD case here and get a client directed flow in e2e https://github.com/Azure/azure-mobile-apps-net-server/blob/master/e2etest/Controllers/Table/AuthenticatedController.cs#L52?

**Comment by brettsam** [comment]
I believe so, yes.

File: test/Microsoft.Azure.Mobile.Server.Authentication.Test/IPrincipalExtensionsTests.cs [hunk 1]
```diff
@@ -161,7 +162,8 @@ public void PopulateProviderCredentials_AzureActiveDirectory_CreatesExpectedCred
 
             IPrincipalExtensions.PopulateProviderCredentials(tokenEntry, credentials);
 
-            Assert.Equal("TestAccessToken", credentials.AccessToken);
+            // id_token should go here, not access_token
+            Assert.Equal("TestIdToken", credentials.AccessToken);
```
Comment (aziel) [inline test/Microsoft.Azure.Mobile.Server.Authentication.Test/IPrincipalExtensionsTests.cs]: this looks weird with the above comment.

Comment (brettsam) [inline test/Microsoft.Azure.Mobile.Server.Authentication.Test/IPrincipalExtensionsTests.cs]: Does this make more sense? "For AAD, the id_token value needs to get copied to AccessToken, not the access_token value."

Comment (aziel) [inline test/Microsoft.Azure.Mobile.Server.Authentication.Test/IPrincipalExtensionsTests.cs]: sounds good

**Comment by aziel** [comment]
Looks good

**Pull request merged**